### PR TITLE
Replace packages.config with project.json

### PR DESCRIPTION
**Pull request by agocke**
As it sounds. This is a biggie. Note that you'll want to disable NuGet autorestore in VS if you try to build individual projects due to slowness. The setting is in Tools->Options->NuGet Package Manager->General->"Automatically check for missing packages during build in Visual Studio".

### CODE REVIEW

**Comment by agocke** [comment]
@jasonmalinowski @jaredpar @tmat FYI

**Comment by agocke** [comment]
retest this please

**Comment by agocke** [comment]
Just using this branch to test CI builds, btw, I've pushed this branch to upstream as CoreClr to work on while we're bringing the rest of the infrastructure pieces together.

**Comment by agocke** [comment]
retest this please

**Comment by agocke** [comment]
Andy, is this a passing build?

**Comment by agocke** [comment]
Yes, yes it is, Other Andy.

File: BuildAndTest.proj [hunk 1]
```diff
@@ -47,10 +44,20 @@
 
   <Target Name="Test">
     <ItemGroup>
-      <TestAssemblies Include="Binaries\$(Configuration)\**\$(IncludePattern)" />
+      <AssembliesToCopyLocal Include="$(NuGetPackageRoot)\xunit\$(XunitVersion)\*.dll
+        ;$(NuGetPackageRoot)\xunit.assert\2.1.0-beta4-build3109\lib\portable-net45+netcore45+wp8+wpa81\*.dll" />
```
Comment (jaredpar) [inline BuildAndTest.proj]: Could we use `$(XunitVersion)` in the path here?

File: build/ToolsetPackages/project.lock.json [hunk 2]
```diff
@@ -0,0 +1,1171 @@
...
+        "lib/tools/System.Collections.Immutable.dll",
+        "lib/tools/System.Reflection.Metadata.dll",
+        "lib/tools/VBCSCompiler.exe",
+        "lib/tools/VBCSCompiler.exe.config",
+        "lib/tools/csc.exe",
+        "lib/tools/csc2.exe",
+        "lib/tools/vbc.exe",
+        "lib/tools/vbc2.exe",
+        "src/Framework/AssemblyInfo.cs",
+        "src/Framework/BuildEngineResult.cs",
```
Comment (jaredpar) [inline build/ToolsetPackages/project.lock.json]: Did they really mean to put the source of MSBuild in the Nuget?

**Comment by jaredpar** [comment]
:+1:

File: build/ToolsetPackages/project.lock.json (hunk 2 above)
Comment (agocke) [inline build/ToolsetPackages/project.lock.json]: You mean the compiler assemblies? Yeah that's not great -- we should try and get them to integrate the Microsoft.Net.Compilers package for this.

**Comment by agocke** [comment]
retest this please

File: build/ToolsetPackages/project.lock.json (hunk 2 above)
Comment (jaredpar) [inline build/ToolsetPackages/project.lock.json]: @agocke no.  Look just above my comment, and the hundreds of entries below.  It appears that the entire source code of MSBuild is contained in their NuGet package.

Comment (agocke) [inline build/ToolsetPackages/project.lock.json]: Huh. Yes, you're right. @AndyGerlicher, thoughts?

**Pull request merged**